src/FitterBase.cc

```diff
@@ -299,5 +305,18 @@ void FitterBase::saveChi2Contributions(std::string const &baseName) const {
     saveChi2RefContributions(refTuple);
 }
 
+float FitterBase::_lineSearch(Eigen::VectorXd const &delta) {
+    auto func = [this, &delta](double scale) {
+        offsetParams(scale * delta);
+        auto chi2 = computeChi2();
+        // reset the system to where it was before offsetting.
+        offsetParams(-scale * delta);
+        return chi2.chi2;
+    };
+    auto result = boost::math::tools::brent_find_minima(func, -1.0, 2.0, std::numeric_limits<double>::digits);
```

Comment:
Note that in principle, the minima can not be located to greater accuracy than the square root of machine epsilon (for 64-bit double, sqrt(1e-16)≅1e-8), therefore the value of bits will be ignored if it's greater than half the number of bits in the mantissa of T.
Since `::digits` gets you the latter this will be ignored. Did you intend that? If so please add a note.

Reply:
[Their example](https://www.boost.org/doc/libs/1_65_0/libs/math/doc/html/math_toolkit/roots/brent_minima.html#math_toolkit.roots.brent_minima.example) does exactly this: 

> For simplicity, we set the precision parameter bits to std::numeric_limits<double>::digits, which is effectively the maximum possible i.e. std::numeric_limits<double>::digits/2.

Given that, and the lack of clarity on what "ignored" means in this context (I think the implication is "we will use the maximum possible precision), I changed it to `digits/2` and also added a note.